tozny/resource_realm.go

```diff
@@ -227,3 +226,40 @@ func resourceRealmDelete(ctx context.Context, d *schema.ResourceData, m interfac
 
 	return diags
 }
+
+func resourceRealmUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	toznySDK, err := MakeToznySDK(d, m)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Check if MPC settings have changed
+	if d.HasChange("mpc_enabled") {
+		mpc_setting, _ := d.Get("mpc_enabled").(bool)
+
+		err = toznySDK.RealmSettingsUpdate(ctx, d.Get("realm_name").(string), identityClient.RealmSettingsUpdateRequest{
```

Comment:
I could probably write this in a cleaner fashion that makes it so that only once call to `RealmSettingsUpdate` is made at the end of this. Lemme know what you think if that should be done

Reply:
If you know better, do better is my vacation level advice ;-)

Reply:
Ya got me, I shall do better 😆

Reply:
yes, prefer a single api call where these if blocks build a `RealmSettingsUpdateRequest` that gets launched (if necessary) at the end. beyond the small performance boost, it will prevent partial successes (eg. imagine what happens if the update to federation fails in the current code. mpc_enabled could successfully update, but the overall terraform update would fail)